hnn_core/tests/test_gui.py

```diff
@@ -413,3 +417,57 @@ def test_gui_adaptive_spectrogram():
                 for attr in dir(gui.viz_manager.figs[figid])]) is False
     assert len(gui.viz_manager.figs[1].axes) == 2
     plt.close('all')
+
+
+def test_unlink_relink_widget():
+    """Tests the unlinking and relinking of widgets decorator."""
+
+    # Create a basic version of the VizManager class
+    class MiniViz:
+        def __init__(self):
+            self.tab_group_1 = Tab()
+            self.tab_group_2 = Tab()
+            self.tab_link = link(
+                (self.tab_group_1, 'selected_index'),
+                (self.tab_group_2, 'selected_index'),
+            )
+
+        def add_child(self, to_add=1):
+            n_tabs = len(self.tab_group_2.children) + to_add
+            # Add figure tab and select latest tab
+            self.tab_group_1.children = \
+                [Text(f'Test{s}') for s in np.arange(n_tabs)]
+            self.tab_group_1.selected_index = n_tabs - 1
+
+            self.tab_group_2.children = \
+                [Text(f'Test{s}') for s in np.arange(n_tabs)]
+            self.tab_group_2.selected_index = n_tabs - 1
+
+        @unlink_relink(attribute='tab_link')
+        def add_child_decorated(self, to_add):
+            self.add_child(to_add)
+
+    # Check that widgets are linked.
+    gui = MiniViz()
+    error = False
+    try:
```

Comment:
Will this work in a `pytest` context manager? Something like

https://github.com/jonescompneurolab/hnn-core/blob/34d972ce8a57d4a04a281a312884f7daa597ec49/hnn_core/tests/test_gui.py#L103C18-L103C18

Reply:
Ooo I didn't know about that. That's much cleaner thanks!